bigquery_etl/cli/utils.py

```diff
@@ -251,3 +251,42 @@ def temp_dataset_option(
         help="Dataset where intermediate query results will be temporarily stored, "
         "formatted as PROJECT_ID.DATASET_ID",
     )
+
+
+def extract_last_group_by_from_query(query_file=None, query=None):
+    """Return the list of columns in the latest group by of a query."""
```

Comment:
Could you expand a little bit what this is used for? 
I'm not sure I understand why ;)